pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -98,13 +128,29 @@ type DockercfgController struct {
 
 	serviceAccountCache      MutationCache
 	serviceAccountController *framework.Controller
+	secretsCache             cache.Store
+	secretController         *framework.Controller
 
 	queue workqueue.RateLimitingInterface
 
 	// syncHandler does the work. It's factored out for unit testing
 	syncHandler func(serviceKey string) error
 }
 
+func (e *DockercfgController) handleTokenPopulation(_, newObj interface{}) {
+	secret := newObj.(*api.Secret)
+	if len(secret.Annotations) == 0 {
+		return
+	}
+	serviceAccountName, _ := secret.Annotations[api.ServiceAccountNameKey]
```

Comment:
this should only pay attention to tokens that this controller created, not any arbitrary token for this service account